app/core/api/views.py

```diff
@@ -116,3 +116,33 @@ class RecurringEventViewSet(viewsets.ModelViewSet):
     permission_classes = [IsAuthenticated]
     queryset = RecurringEvent.objects.all()
     serializer_class = RecurringEventSerializer
+
+
+@extend_schema_view(
+    list=extend_schema(description="Return a list of all the sponsor partners"),
+    create=extend_schema(description="Create a new sponsor partner"),
+    retrieve=extend_schema(description="Return the details of a sponsor partner"),
+    destroy=extend_schema(description="Delete a sponsor partner"),
+    update=extend_schema(description="Update a sponsor partner"),
+    partial_update=extend_schema(description="Patch a sponsor partner"),
+)
+class SponsorPartnerViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    queryset = SponsorPartner.objects.all()
+    serializer_class = SponsorPartnerSerializer
+
+    def get_queryset(self):
```

Comment:
Good work anticipating the need. I think I would skip it in future models so we can get all the models done sooner. There's that tradeoff between doing this now and waiting until a client tells us they need it, of if we run short of things to do later.

This is also a good opportunity to add more documentation.
I think this function override might trigger the code coverage tool complain about this code being untested. You will have to add a test for this. See `test_get_single_user()` in `test_api.py` for something similar.
Modify in `app/setup.cfg` the `pytest` section to have `--cov-report html` instead of terminal to generate html output. Run `test.sh`, then view `app/htmlcov/index.html` in a browser or right-click on it in vscode and choose Open with live server. Click into `views.py` and scroll down to see the lines it thinks are untested.
It's not great to test multiple things in one test like I did there. If you want to play with the `parametrize` feature in pytest, this could be an opportunity. Search for `parametrize` in the test code for an example.

Reply:
Sorry, I'm confused here--if I leave out the `get_queryset()` function, that additional test doesn't need to be run, correct? Or would you like me to leave in the `get_queryset()` in this case and add the new test?

Reply:
Either way is fine. Leaving out the `get_queryset()` will satisfy the code coverage tool.

I was suggesting adding tests and all that as extra practice in case you feel really attached to this code. I don't know if it's too overwhelming at this point since everything else is already sort of new to you.

I think someone will make use of the code though.

Reply:
I would ultimately like to get that extra practice and familiarity with writing tests, but for now I don't think I have the bandwidth. I'm just going to comment out the `get_queryset()` code, and later on someone can uncomment and use it if they like.